openfisca_aotearoa/variables/acts/social_security/supported_living_payment/supported_living_payment.py

```diff
@@ -1,124 +1,236 @@
+    reference = [
+        "https://www.legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783174",
+        ]
 
+    def formula_2018_11_26(persons, period, parameters):
+        # person P has restricted work capacity or is totally blind
+        ssa34_a = persons("supported_living_payment__restricted_work_capacity", period.first_week) + persons("totally_blind", period.start)
 
-class supported_living_payment__caring_for_another_person(variables.Variable):
+        # person P meets the residential requirement
+        ssa34_b = persons("social_security__residential_requirement", period.first_week)
+
+        # person P is 16 or older
+        ssa34_c = persons("age", period.start) >= 16
+
+        # ssa35 specifies requirements for restricted work capacity status
+
+        # person P is ineligible if the disability was self-inflicted to gain benefits
+        ssa36 = numpy.logical_not(persons("supported_living_payment__disability_self_inflicted", period))
+
+        # ssa37 specifies medical examination for reassessment of work capacity
+
+        # ssa38 specifies payment rates for person P and partner / spouse S if applicable
+
+        # ssa39 specifies open employment is permitted for reassessment of work capacity
+
+        return ssa34_a * ssa34_b * ssa34_c * ssa36
+
+    def formula(persons, period, parameters):
+        # person P has restricted work capacity
+        ssa40B_1_a = persons("supported_living_payment__restricted_work_capacity", period.first_week)
+
+        # person P is totally blind
+        ssa40B_1_b = persons("totally_blind", period)
+
+        # person P is 16 or older
+        ssa40B_1A = persons("age", period.start) >= 16
+
+        # person P meets the residential requirement
+        ssa40B_1B = persons("social_security__residential_requirement", period.first_week)
+
+        ssa40B_1 = ssa40B_1A * ssa40B_1B * (ssa40B_1_a + ssa40B_1_b)
+
+        # 40B_2 specifies requirements for restricted work capacity status
+
+        # 40B_3 specifies requirements for restricted work capacity status
+
+        # 40B_4 (redundantly?) states that person P must be either totally blind or permanently disabled
+
+        # person P is ineligible if the disability was self-inflicted to gain benefits
+        ssa40B_5 = numpy.logical_not(persons("supported_living_payment__disability_self_inflicted", period))
+
+        return ssa40B_1 * ssa40B_5
+
+
+class supported_living_payment__carer__entitled(variables.Variable):
     value_type = bool
     entity = entities.Person
     definition_period = periods.WEEK
-    label = "Eligible for Supported Living Payment."
-    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783187"
-    set_input = holders.set_input_dispatch_by_period
+    label = "Eligible for Supported Living Payment, on grounds of caring for another person."
+    reference = [
+        "https://www.legislation.govt.nz/act/public/2018/0032/latest/whole.html#DLM6783187",
```

Comment:
Add references to the relevant versions of the1964 Act:
https://www.legislation.govt.nz/act/public/1964/0136/219.0/DLM5469766.html
https://www.legislation.govt.nz/act/public/1964/0136/334.0/DLM5469766.html